### PR TITLE
feat: Map OIDC scope to Presentation Exchange

### LINKED ISSUE #5: Map OIDC scope to Presentation Exchange
https://identity.foundation/presentation-exchange/

https://github.com/decentralized-identity/presentation-exchange/pull/14/files#diff-9308a2575ae14de31b25f4459ecfc604R437

### PR DESCRIPTION
**Pull request by fqutishat**
closes #5

Signed-off-by: Firas Qutishat <firas.qutishat@securekey.com>

### CODE REVIEW

File: cmd/adapter-rest/startcmd/start.go [hunk 1]
```diff
@@ -134,13 +140,20 @@ func getAdapterRestParameters(cmd *cobra.Command) (*adapterRestParameters, error
...
-		dbURL:             dbURL,
-		oidcProviderURL:   oidcURL,
-		staticFiles:       staticFiles,
+		hostURL:                     hostURL,
+		tlsSystemCertPool:           tlsSystemCertPool,
+		tlsCACerts:                  tlsCACerts,
+		dbURL:                       dbURL,
+		oidcProviderURL:             oidcURL,
+		staticFiles:                 staticFiles,
+		presentationDefinitionsFile: presentationDefinitionsFile,
```
Comment (llorllale) [inline cmd/adapter-rest/startcmd/start.go:156]: we should start thinking about saving this config to the database, and exposing an endpoint for this configuration.

File: pkg/restapi/rp/operation/operations.go [hunk 2]
```diff
@@ -80,10 +94,27 @@ func (o *Operation) hydraConsentHandler(w http.ResponseWriter, _ *http.Request)
 	testResponse(w)
 }
 
-// Frontend requests a request for a presentation and provides a handle.
-func (o *Operation) getPresentationRequestHandler(w http.ResponseWriter, _ *http.Request) {
-	// TODO extract handle and return the request for presentation.
-	testResponse(w)
+// Frontend requests to create presentation definition.
+func (o *Operation) createPresentationDefinition(rw http.ResponseWriter, req *http.Request) {
```
Comment (llorllale) [inline pkg/restapi/rp/operation/operations.go:98]: This endpoint is called by the frontend. The user is redirected to the frontend in the consent phase, and the backend already has the scopes. It doesn't make sense to pass the scopes to the frontend for it to then pass it back to the backend to create the presentation. It also opens up this endpoint to replay exploits, etc. The idea here was to have the backend create a unique handle of one-time use to a presentation, and pass that to the frontend. The frontend would call the backend with this handle and get the presentation.

Comment (fqutishat) [reply pkg/restapi/rp/operation/operations.go:98]: I didn't know about this unique handle

Comment (fqutishat) [reply pkg/restapi/rp/operation/operations.go:98]: @llorllale who is working on unique handle

Comment (fqutishat) [reply pkg/restapi/rp/operation/operations.go:98]: @troyronda @sudeshrshetty @rolsonquadras ^^^^

Comment (llorllale) [reply pkg/restapi/rp/operation/operations.go:98]: @fqutishat it is in the TODO comment delete in this PR, and also in `hydraConsentHandler`

Comment (llorllale) [reply pkg/restapi/rp/operation/operations.go:98]: @fqutishat see #14

Comment (fqutishat) [reply pkg/restapi/rp/operation/operations.go:98]: @llorllale I add TODO #14

**Pull request merged**